independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/CreateProjectCommandHandler.java

```diff
@@ -20,10 +21,8 @@
 
     @Override
     public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws QuarkusCommandException {
-        final ProjectWriter projectWriter = invocation.getProjectWriter();
-        if (projectWriter == null) {
-            throw new IllegalStateException("Project writer has not been provided");
-        }
+        final ProjectWriter projectWriter = new FileProjectWriter(
+                invocation.getQuarkusProject().getProjectFolderPath().toFile());
```

Comment:
Just a note, previously a single invocation could have been passed to multiple handlers (e.g. create followed by add extensions) that would use the same project writer instance. It doesn't mean it has to stay that way.

Reply:
I am not there yet, but to make it more consistent and simple, my intent is to go toward atomic operations (without having to deal with the writer close/save from outside):
- create (directly with or without extensions)
- add extension(s)
- remove extension(s)

Using the same writer would make it more performant when doing multiple of those, but I guess there is no use case for it.

One good example of the risk of having this close/save handled externally is that (as I did without knowing) having multiple writer and closing the one without the modification after ending up with reseting the modification 😅

WDYT?